### ISSUE TITLE
Support GUI signs

### ISSUE THREAD

**Issue by equalsraf** [neovim/neovim#4143]
`sign` support allowed the GUI to place icons as defined through the `sign icon= command, but we have no equivalent for remote GUIs, and the old code got removed.

Ping equalsraf/neovim-qt#90.

**Comment by justinmk** [comment]
Yeah, the old code was not something I think we'd want to bring over. Perhaps we can allow arbitrary msgpack objects to be stored on a sign? Then UIs can use them how they like. Plugins would need to use API calls to "configure" the signs; we don't want to extend the `:sign` command.
